addons/stock_account/stock_account.py

```diff
@@ -31,7 +31,11 @@ class account_invoice_line(osv.osv):
 
     def _get_anglo_saxon_price_unit(self):
         self.ensure_one()
-        return self.product_id.standard_price
+        price = self.product_id.standard_price
+        if not self.uom_id or self.product_id.uom_id == self.uom_id:
+            return price
+        else:
+            return self.product_id.uom_id._compute_price(self.product_id.uom_id.id, price, to_uom_id=self.uom_id.id)
```

Comment:
I am not sure it will work correctly in all cases as this function is inherited and some of these functions might suppose it is in the UoM of the product.